### PR TITLE
basic api fucntionality

### PR DESCRIPTION
**Pull request by SuadHus**
pleeeeasssee acceppppt

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,8 +6,13 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
 
 include_directories(include)
 
-file(GLOB_RECURSE SRC_FILES src/*.cpp)
```
Comment (mohammed-shakir) [inline CMakeLists.txt:9]: dont remove this

File: CMakeLists.txt [hunk 2]
```diff
@@ -6,8 +6,13 @@ set(CMAKE_CXX_STANDARD_REQUIRED True)
...
+# Find libcurl
+find_package(CURL REQUIRED)
 
-add_executable(SOEP ${SRC_FILES})
+# Find nlohmann_json
+find_package(nlohmann_json 3.2.0 REQUIRED)
 
-target_compile_options(SOEP PRIVATE -Wall)
\ No newline at end of file
+add_executable(SOEP src/main.cpp)
```
Comment (mohammed-shakir) [inline CMakeLists.txt:15]: compile the whole source tree

```
add_executable(SOEP ${SRC_FILES})
```

File: src/main.cpp
Comment (mohammed-shakir) [inline src/main.cpp:1]: Dont add this code in main, create a seperate node that makes the API call

**Review by mohammed-shakir — CHANGES_REQUESTED** [review]
Some changes needed, but code looks good overall

File: README.md
Comment (mohammed-shakir) [inline README.md:1]: Add in the README, where to find the .env file

File: README.md [hunk 3]
```diff
@@ -16,5 +16,5 @@ Docker
 
 ```
 docker build -t soep .
-docker run --rm soep
+docker run --rm -e N2YO_API_KEY="<env_key>" soep
```
Comment (mohammed-shakir) [inline README.md:19]: Add in the README, where to find the .env file

**Review by syko240 — COMMENTED** [review]
Is this for the whole [issue](https://github.com/SOEP-Group/SOEP/issues/2)?

Does not really align with what is described in the issue (create a networking module). My proposal: crate a generic API request functionality which returns the a buffer of the raw JSON string. Code messy, improve file structure or somthin.
like:
```
src/
    main.cpp
    (network/api) /
         api_call.h
         api_call.cpp
```
         
         
     
or something similar, up to repo owner
just please dont throw everything in main ?_?

**Comment by mohammed-shakir** [comment]
> Is this for the whole [issue](https://github.com/SOEP-Group/SOEP/issues/2)?
> 
> Does not really align with what is described in the issue (create a networking module). My proposal: crate a generic API request functionality which returns the a buffer of the raw JSON string. Code messy, improve file structure or somthin. like:
> 
> ```
> src/
>     main.cpp
>     (network/api) /
>          api_call.h
>          api_call.cpp
> ```
> 
> or something similar idgaf, up to repo owner just please dont throw everything in main ?_?

yeah this structure that you suggested should be good enough for now. But the issue is a bit missleading, it was supposed to only be basic API call to fetch the into of 1 satellite (for now), and then @SuadHus was going to build upon that by creating more issues.

I suggest you (@SuadHus ) rewrite the issue to match what you have done and then create more issues for the rest of the functionallities mentioned in this issue.

@syko240 don't swear in your pr reviews 💀

**Comment by SuadHus** [comment]
I didn't close the issue jusr pushed

File: README.md (hunk 3 above)
Comment (SuadHus) [reply README.md:19]: what??? Are you stupid?

**Comment by SuadHus** [comment]
PLEASSSE ACCEPT

**Review by syko240 — COMMENTED** [review]
LG, just made one change.

**Review by mohammed-shakir — APPROVED** [review]
LGTM!

**Pull request merged**